spring-data-rest-webmvc/src/main/java/org/springframework/data/rest/webmvc/json/DomainObjectReader.java

```diff
@@ -240,13 +241,18 @@ private boolean handleArrayNode(ArrayNode array, Collection<Object> collection,
 		for (JsonNode jsonNode : array) {
 
 			if (!value.hasNext()) {
+				if (componentType != null) {
+					collection.add(mapper.treeToValue(jsonNode, componentType));
+				}
```

Comment:
I think we have to extend the algorithm here as this will only work if there's exactly *one* more item provided. It will get added but then the iteration is stopped so that all subsequent incoming elements are ignored, right? I guess we could just trigger a `continue` if there are more elements in the array only resorting to the eventual `return` if we have maxed out the array elements.

I am kind of wondering whether we could or should actually invert the processing here: iterating over the existing  elements first, removing items if we max out the array first and resort to adding everything thats left in the array after that?